xcframeworks.sh

```diff
@@ -0,0 +1,174 @@
+    -framework "${BUILD_PATH}/${IOS_SIM_ARCHIVE}/Products/Library/Frameworks/${1}.framework" \
+    -framework "${BUILD_PATH}/${IOS_ARCHIVE}/Products/Library/Frameworks/${1}.framework" \
+    -framework "${BUILD_PATH}/${TVOS_SIM_ARCHIVE}/Products/Library/Frameworks/${1}.framework" \
+    -framework "${BUILD_PATH}/${TVOS_ARCHIVE}/Products/Library/Frameworks/${1}.framework" \
+    -framework "${BUILD_PATH}/${WATCHOS_SIM_ARCHIVE}/Products/Library/Frameworks/${1}.framework" \
+    -framework "${BUILD_PATH}/${WATCHOS_ARCHIVE}/Products/Library/Frameworks/${1}.framework" \
+    -framework "${BUILD_PATH}/${MACOS_ARCHIVE}/Products/Library/Frameworks/${1}.framework" \
+    -output "${XCFRAMEWORK_PATH}/${1}".xcframework;
+}
+
+# create xcframework for each product that contains all supported platforms
+function create_xcframework {   
+    if [[ "$IOS_ONLY_PRODUCTS" == *"$1"* ]]; then
+        create_xcframework_ios_only "${1}"   
+    else
+        create_xcframework_all "${1}"
+    fi
+}
+
+# create archives for ios only products
+function create_archives_ios_only {
+    archive "$1" "${IOS_SIM_DESTINATION}" "${BUILD_PATH}/${IOS_SIM_ARCHIVE}" "iphonesimulator"
+    archive "$1" "${IOS_DESTINATION}" "${BUILD_PATH}/${IOS_ARCHIVE}" "iphoneos";
+    create_xcframework "$1"
+}
+
+# create archives for products supporting all platforms
+function create_archives_all {
+    archive "$1" "${IOS_SIM_DESTINATION}" "${BUILD_PATH}/${IOS_SIM_ARCHIVE}" "iphonesimulator"
+    archive "$1" "${IOS_DESTINATION}" "${BUILD_PATH}/${IOS_ARCHIVE}" "iphoneos"
+    archive "$1" "${TVOS_SIM_DESTINATION}" "${BUILD_PATH}/${TVOS_SIM_ARCHIVE}" "appletvsimulator"
+    archive "$1" "${TVOS_DESTINATION}" "${BUILD_PATH}/${TVOS_ARCHIVE}" "appletvos"
+    archive "$1" "${WATCHOS_SIM_DESTINATION}" "${BUILD_PATH}/${WATCHOS_SIM_ARCHIVE}" "watchsimulator"
+    archive "$1" "${WATCHOS_DESTINATION}" "${BUILD_PATH}/${WATCHOS_ARCHIVE}" "watchos"
+    archive "$1" "${MACOS_DESTINATION}" "${BUILD_PATH}/${MACOS_ARCHIVE}" "macosx"
+    create_xcframework "$1" 
+}
+
+# create all archives for each product and platform
+function create_archives {
+    if [ -z "$1" ]
+      then
+        echo "WARNING, NO VERSION NUMBER ENTERED. ASSUMING LATEST."
+        define_product_name "${LATEST_MAJOR}"
+    else
+        define_product_name "$1"
+    fi
+    for i in "${PRODUCT_NAME[@]}"; 
+        do
+            if [[ "$IOS_ONLY_PRODUCTS" == *"$i"* ]]; then                
+                create_archives_ios_only "$i"
+            else
+                create_archives_all "$i" 
+            fi    
+        done    
+}
+
+# zip all the xcframeworks
+function zip_xcframeworks {
+    if [[ -d "${XCFRAMEWORK_PATH}" ]]; then
+        zip -r "${ZIP_PATH}" "${XCFRAMEWORK_PATH}"
+        rm -rf "${XCFRAMEWORK_PATH}"
+    fi
+}
+
+#### start ####
+clean_build_folder
+
+# do the work
+create_archives "$1"
+zip_xcframeworks
+
+
+echo ""
+echo "Done! Upload ${ZIP_PATH} to GitHub when you create the release."
```

Comment:
We should really make a more generic script for single scheme projects that you can just pass a scheme name and project path. Not really needed now but we should keep that in mind for the future